cl/recap/tasks.py

```diff
@@ -3435,3 +3437,89 @@ def do_recap_document_fetch(epq: EmailProcessingQueue, user: User) -> None:
         process_recap_email.si(epq.pk, user.pk),
         extract_recap_pdf.s(),
     ).apply_async()
+
+
+def _run_availability_check(rd: RECAPDocument) -> bool:
+    """Helper function for fetch_prayer_availability().
+
+    :param rd: The RECAPDocument of interest
+    :return: bool that indicates whether document is available
+    """
+    court_id = rd.docket_entry.docket.court.pk
+    pacer_doc_id = rd.pacer_doc_id
+
+    recap_user = User.objects.get(username="recap")
+    session_data = get_or_cache_pacer_cookies(
+        recap_user.pk, settings.PACER_USERNAME, settings.PACER_PASSWORD
+    )
+    s = ProxyPacerSession(
+        cookies=session_data.cookies, proxy=session_data.proxy_address
+    )
+    receipt_report = DownloadConfirmationPage(court_id, s)
+    receipt_report.query(pacer_doc_id)
+    data = receipt_report.data
+
+    if data == {}:
+        rd.is_sealed = True
+        rd.save()
+        PrayerAvailability.objects.update_or_create(
+            recap_document=rd, defaults={"last_checked": now}
+        )
+        return False
+    else:
+        # making sure that previously sealed documents that are now available are marked as such
+        rd.is_sealed = False
+        rd.save()
+        PrayerAvailability.objects.filter(recap_document=rd).delete()
+
+    if data.billable_pages != 30:
+        rd.page_count = data.billable_pages
+        rd.save()
+
+    return True
+
+
+@app.task(
+    bind=True,
+    autoretry_for=(RedisConnectionError, PacerLoginException),
+    max_retries=5,
+    interval_start=5,
+    interval_step=5,
+    ignore_result=True,
+)
+@transaction.atomic
+def fetch_prayer_availability(self, pk: int) -> bool:
```

Comment:
Let's refactor this method. We can change it to a regular function and remove the call to _run_availability_check. The function should simply return a boolean value, indicating whether or not we need to schedule a task to check document availability. Here's a suggestion for the signature:

```suggestion
def should_check_prayer_availability(self, pk: int) -> bool:
```